boards/recovery/src/main.rs

```diff
@@ -136,6 +144,28 @@ mod app {
         loop {}
     }
 
+    // handle recovery counter
+    // start a counter
+    #[task(local = [recovery_timer])]
+    fn recovery_counter_tick(mut cx: recovery_counter_tick::Context) {
+        let timer = cx.local.recovery_timer;
+        // should probably check if timer is already running
+        // to avoid resetting it
+        
+        let duration_mins = atsamd_hal::fugit::MinutesDurationU32::minutes(1);
+        // timer requires specific duration format
+        let timer_duration: atsamd_hal::fugit::Duration<u32, 1, 1000000000> = duration_mins.convert();
+        timer.start(timer_duration);
+    }
+
+    // interrupt handler for when counter is finished
+    #[task(binds=TC2, shared=[data_manager])]
+    fn recovery_counter_finish(mut cx: recovery_counter_finish::Context) {
```

Comment:
You may need to clear the interrupt flag so that this can fire again. You should also be checking if cx.local.timer.wait().is_ok(). The reason is that another interrupt associated with the timer could fire and call this funciton which doesn't mean that the timer is actually done. This means that you should move the timer to the shared struct.